tests/integration_test_net.go

```diff
@@ -288,6 +288,79 @@ func (n *IntegrationTestNet) GetClient() (*ethclient.Client, error) {
 	return ethclient.Dial("http://localhost:18545")
 }
 
+// RestartWithExportImport stops the network, exports the genesis file, cleans the
+// temporary directory, imports the genesis file, and starts the network again.
+func (n *IntegrationTestNet) RestartWithExportImport() error {
+	n.Stop()
+	fmt.Println("Network stopped. Exporting genesis file...")
+
+	// save original args
+	originalArgs := os.Args
+
+	// export
+	os.Args = []string{
+		"sonictool",
+		"--datadir", n.directory,
+		"genesis", "export", "testGenesis.g",
```

Comment:
It would be cleaner to create the genesis file in a temporary directory to (a) avoid the file to be staying around after the test and (b) to avoid collisions if this function is called concurrently.

Idea: move the client's state into a sub-directory (`filepath.Join(n.directory, "state")`) and initialize the state there for regular runs. Produce the genesis file in the `n.directory` and remove and re-create the state directory inside the `n.directory`.